presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java

```diff
@@ -69,16 +70,16 @@ protected void validateRootGroups(ManagerSpec managerSpec)
             if (group.getSchedulingPolicy().isPresent()) {
                 switch (group.getSchedulingPolicy().get()) {
                     case WEIGHTED:
-                        for (ResourceGroupSpec subGroup : group.getSubGroups()) {
-                            checkArgument(subGroup.getSchedulingWeight().isPresent(), "Must specify scheduling weight for each sub group when using \"weighted\" scheduling policy");
-                        }
-                        break;
                     case WEIGHTED_FAIR:
+                        if (!(subGroups.stream().allMatch(t -> t.getSchedulingWeight().isPresent()) || subGroups.stream().noneMatch(t -> t.getSchedulingWeight().isPresent()))) {
+                            throw new IllegalArgumentException(format("Must specify scheduling weight for all sub-groups of '%s' or none of them", group.getName()));
```

Comment:
Can we use `checkArgument` to be consistent with the rest of the file?

Reply:
Fixed.